### PR TITLE
Replace usages of `fatalError` that are recoverable by non-fatal error propagation constructs

### PR DESCRIPTION
**Pull request by ahoppen**
Best reviews commit-by-commit.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Review by bnbarham — APPROVED** [review]
Woo, fewer fatals!

File: Sources/SourceKitLSP/Swift/SwiftLanguageService.swift [hunk 1]
```diff
@@ -518,7 +518,12 @@ extension SwiftLanguageService {
       do {
         _ = try await self.sourcekitd.send(req, fileContents: nil)
       } catch {
-        fatalError("failed to apply edit")
```
[the remark this answers is not in the document]
Comment (ahoppen) [reply Sources/SourceKitLSP/Swift/SwiftLanguageService.swift:521]: Yes, I think it’s very unlikely this is going to happen and I don’t think it’s worth optimizing for this case.

File: Sources/SourceKitLSP/Swift/SyntaxHighlightingTokenParser.swift [hunk 2]
```diff
@@ -220,7 +220,8 @@ extension Range where Bound == Position {
     guard let startIndex = snapshot.index(of: lowerBound),
       let endIndex = snapshot.index(of: upperBound)
     else {
-      fatalError("Range \(self) reaches outside of the document")
+      logger.fault("Range \(self) reaches outside of the document")
+      return [self]
```
[the remark this answers is not in the document]
Comment (ahoppen) [reply Sources/SourceKitLSP/Swift/SyntaxHighlightingTokenParser.swift:224]: This made me re-think recovery inside position conversions altogether and I think positions conversions should never fail and always return the closest valid position, which also simplifies all client code. I’ll create a separate PR for that.

[the remark this answers is not in the document]
Comment (ahoppen) [reply Sources/SourceKitLSP/Swift/SyntaxHighlightingTokenParser.swift:224]: And here we go: https://github.com/apple/sourcekit-lsp/pull/1168

**Pull request merged**